src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/ProfessionalUserIntegrationTest.java

```diff
@@ -78,4 +85,58 @@ void shouldInsertOneUserIntoUserRefreshQueue_AndClearOrganisationRefreshQueue()
         assertEquals(0, userRefreshEntity.getRetry());
         assertNotNull(userRefreshEntity.getRetryAfter());
     }
-}
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD,
+            scripts = {"classpath:sql/insert_organisation_profiles.sql"})
+    void shouldRollback_AndUpdateRetryToOneOnException() {
+        ProfessionalUser professionalUser = buildProfessionalUser(1);
+        UsersOrganisationInfo usersOrganisationInfo = buildUsersOrganisationInfo(123, professionalUser);
+        UsersByOrganisationResponse page1 =
+                buildUsersByOrganisationResponse(usersOrganisationInfo, "1", "1", true);
+
+        when(prdService.fetchUsersByOrganisation(any(), eq(null), eq(null), any()))
+                .thenReturn(ResponseEntity.ok(page1));
+
+        // throwing exception on 2nd page retrieval to test rollback (user inserts from page 1 SHOULD be rolled back)
+        when(prdService.fetchUsersByOrganisation(any(), any(String.class), any(String.class), any()))
+                .thenThrow(ServiceException.class);
+
+        professionalUserService.findAndInsertUsersWithStaleOrganisationsIntoRefreshQueue();
+
+        assertEquals(userRefreshQueueRepository.findAll().size(), 0);
```

Comment:
```suggestion
        assertEquals(0, userRefreshQueueRepository.findAll().size());
```

Reply:
suggestion applied in [ce66cc0](https://github.com/hmcts/am-org-role-mapping-service/pull/1788/commits/ce66cc0e2224ee0c89290c061f894bf27e86e0dc)